FBShimmering/FBShimmeringLayer.m

```diff
@@ -244,6 +244,22 @@ - (void)setShimmeringOpacity:(CGFloat)shimmeringOpacity
   }
 }
 
+- (CFTimeInterval)shimmeringBeginTime
+{
+  return [_maskLayer animationForKey:kFBShimmerSlideAnimationKey].beginTime;
+}
+
+- (void)resumeShimmeringWithBeginTime:(CFTimeInterval)beginTime
+{
+  CAAnimation *slideAnimation = [[_maskLayer animationForKey:kFBShimmerSlideAnimationKey] mutableCopy];
+  if (nil == slideAnimation) {
+    return;
+  }
+
+  slideAnimation.beginTime = beginTime;
+  [_maskLayer addAnimation:slideAnimation forKey:kFBShimmerSlideAnimationKey];
```

Comment:
I don’t think any sentinel value does a good job of conveying the default, which is to set the begin time to the current time. If people want to do this, how about just letting them set it to `CACurrentMediaTime()`?

Alternatively, my initial approach works. 😛

> On Apr 28, 2015, at 4:28 PM, Grant Paul <notifications@github.com> wrote:
> 
> In FBShimmering/FBShimmeringLayer.m:
> 
> > @@ -244,6 +244,22 @@ - (void)setShimmeringOpacity:(CGFloat)shimmeringOpacity
> >    }
> >  }
> >  
> > +- (CFTimeInterval)shimmeringBeginTime
> > +{
> > +  return [_maskLayer animationForKey:kFBShimmerSlideAnimationKey].beginTime;
> > +}
> > +
> > +- (void)resumeShimmeringWithBeginTime:(CFTimeInterval)beginTime
> > +{
> > +  CAAnimation *slideAnimation = [[_maskLayer animationForKey:kFBShimmerSlideAnimationKey] mutableCopy];
> > +  if (nil == slideAnimation) {
> > +    return;
> > +  }
> > +
> > +  slideAnimation.beginTime = beginTime;
> > +  [_maskLayer addAnimation:slideAnimation forKey:kFBShimmerSlideAnimationKey];
> Good point, maybe needs a different sentinel value.
> 
> Yeah, very true about CACurrentMediaTime() — non-linear timescales are rare, but good to support them!
> 
> —
> Reply to this email directly or view it on GitHub.
>